android/festago/app/src/main/java/com/festago/festago/presentation/ui/home/HomeViewModel.kt

```diff
@@ -19,13 +19,23 @@ class HomeViewModel @Inject constructor(private val authRepository: AuthReposito
     private val _event = MutableSharedFlow<HomeEvent>()
     val event: SharedFlow<HomeEvent> = _event.asSharedFlow()
 
-    fun loadHomeItem(homeItemType: HomeItemType) {
+    private val _selectedItem = MutableStateFlow(HomeItemType.FESTIVAL_LIST)
+    val selectedItem: StateFlow<HomeItemType> = _selectedItem.asStateFlow()
+
+    fun selectItem(homeItemType: HomeItemType) {
+        when (homeItemType) {
+            HomeItemType.FESTIVAL_LIST -> _selectedItem.value = homeItemType
+            HomeItemType.TICKET_LIST -> selectItemWhenSignIn(HomeItemType.TICKET_LIST)
+            HomeItemType.MY_PAGE -> selectItemWhenSignIn(HomeItemType.MY_PAGE)
+        }
+    }
+
+    private fun selectItemWhenSignIn(homeItemType: HomeItemType) {
         viewModelScope.launch {
-            when {
-                homeItemType == FESTIVAL_LIST -> _event.emit(HomeEvent.ShowFestivalList)
-                !authRepository.isSigned -> _event.emit(HomeEvent.ShowSignIn)
-                homeItemType == TICKET_LIST -> _event.emit(HomeEvent.ShowTicketList)
-                homeItemType == MY_PAGE -> _event.emit(HomeEvent.ShowMyPage)
+            if (!authRepository.isSigned) {
+                _event.emit(HomeEvent.ShowSignIn)
+            } else {
+                _selectedItem.emit(homeItemType)
```

Comment:
```suggestion
            if (authRepository.isSigned) {
                _selectedItem.emit(homeItemType)
            } else {
                _event.emit(HomeEvent.ShowSignIn)
```
흐름상 `!`을 빼도 괜찮을 것같은데 어떻게 생각하시나요??